test/integrations/component.test.ts

```diff
@@ -34,6 +38,68 @@ afterAll(async () => {
   await createHttpTerminator({ server }).terminate();
 });
 
+// unmock already existing axios-mocking
+jest.unmock('axios')
+
+jest.mock('axios')
+const formAxiosReqsMap = (calls: MockHttpCallsData[]) => {
+  try {
+    return calls.reduce((agg, curr) => {
+      let obj = curr.httpReq;
+      return { ...agg, [stringify(obj)]: curr.httpRes };
+    }, {})
+  } catch (error) {
+    return {}
+  }
+}
+
+const mockImpl = (type, axReqMap) => {
+  // return value fn
+  const retVal = (key) => {
+    if (axReqMap[key]) {
+      return axReqMap[key];
+    }
+    return {
+      status: 500,
+      body: 'Something bad'
+    }
+  }
+
+  if (['constructor'].includes(type)) {
+    return (opts) => {
+      // mock result from some cache
+      const key = stringify({ ...opts })
+      return retVal(key);
+    };
+  } else if (['delete', 'get'].includes(type)) {
+    return (url, opts) => {
+      // mock result from some cache
+      const key = stringify({ url, ...opts})
+      return retVal(key);
+    };
+  }
+  
+  // post, patch, put
+  return (url, data, opts) => {
+    // mock result from some cache
+    const key = stringify({ url, data, ...opts })
+    return retVal(key);
+  };
```

Comment:
Can't we have one common function apart from three? Which will stringify url, data and opts always.

Reply:
Will try to see if something of that sort is possible.

Reply:
The approach has been changed. Hence resolving this comment